src/annmas/segment/command.py

```diff
@@ -31,13 +33,48 @@ def to_tag(self):
 
 @click.command(name="segment")
 @click_log.simple_verbosity_option(logger)
-@click.option("-m", "--model", required=False, type=click.Path(exists=True), help="pre-trained model to apply")
-@click.option("-t", "--threads", type=int, default=1, show_default=True, help="number of threads to use (0 for all)")
-@click.option("-o", "--output-bam", required=True, type=click.Path(exists=False), help="segmented bam output")
-@click.argument('input-bam', type=click.Path(exists=True))
-def main(model, threads, output_bam, input_bam):
+@click.option(
+    "-m",
+    "--model",
+    required=False,
+    type=click.Path(exists=True),
+    help="pre-trained model to apply",
+)
+@click.option(
+    "-t",
+    "--threads",
+    type=int,
+    default=mp.cpu_count() - 1,
+    show_default=True,
+    help="number of threads to use (0 for all)",
+)
+@click.option(
+    "-o",
+    "--output-bam",
+    required=True,
+    type=click.Path(exists=False),
+    help="segment-annotated bam output",
+)
+@click.option(
+    "-p",
+    "--split-bam",
+    required=False,
+    type=click.Path(exists=False),
+    help="segmented bam output",
+)
+@click.option(
+    "-s",
+    "--do-simple-splitting",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Do splitting of reads based on splitter delimiters, rather than whole array structure."
+    "This splitting will cause delimiter sequences to be repeated in each read they bound.",
```

Comment:
This first implementation was basically porting the existing splitters to this codebase.  I wanted to preserve functionality so I could just slot it into the existing pipeline.

I can update it to not include the delimiters in the actual sequences.  For now I'll have it keep the annotations for all segments so we can see how each read was segmented.

Reply:
The fix for this was harder for the simple splitting than I expected it to be, but I put one in.  Right now it basically removes everything up to the first `random` sequence (though the first and last sequences are split slightly differently because of the delimiters used).